Refactor to use dataclasses

MichaelClifford opened the pull request:
This is a WIP PR, where most of the flatliners have been updated to input and output dataclass's instead of dictionaries.

Commit:
WIP dataclasses

Commit:
WIP dataclasses

durandom commented on flatliners/comparisonscore.py:
```diff
@@ -53,16 +51,27 @@ def compute_cluster_distance(self, values):
+@dataclass
+class STD_COMPARISON_DATA:
+
+    cluster: str = ""
```
I think we should ignore E701 for those dataclasses:
Add this to .stickler.yml in this PR and we should be good

```diff
diff --git a/.stickler.yml b/.stickler.yml
index ae58708..167182d 100644
--- a/.stickler.yml
+++ b/.stickler.yml
@@ -3,9 +3,11 @@ linters:
     python: 3
     max-line-length: 160
     fixer: true
+    ignore: 'E701'
   pep8:
     python: 3
     max-line-length: 160
     fixer: true
+    ignore: 'E701'
 fixers:
   enable: true
```

durandom commented on flatliners/comparisonscore.py:
```diff
@@ -53,16 +51,27 @@ def compute_cluster_distance(self, values):
               return False
 
+@dataclass
+class STD_COMPARISON_DATA:
```
why UPPER_CASE? 
And drop the `DATA` suffix.

durandom reviewed (commented):
Maybe we'll nest the dataclasses **inside** the flatliner? This way we could always name them the same. E.g. `Data`, although `Data` is too generic, maybe you can come up with a better name? `Signal`, `Cue`, `Product`, `Result`?

Like this:

```python
class Human:

    def __init__(self):
        self.name = 'Guido'
        self.head = self.Head()

    class Head:
        def talk(self):
            return 'talking...'

if __name__ == '__main__':
    guido = Human()
    print (guido.name)
    print (guido.head.talk())
    head = Human.Head()
    print(head.talk())
```

this from https://pythonspot.com/inner-classes/

Commit:
dataclass changes

Commit:
dataclass changes

MichaelClifford committed:
Merge branch 'master' into data_classes

Commit:
dataclass tests

Commit:
dataclass tests

Commit:
Merge branch 'data_classes' of https://github.com/MichaelClifford/prometheus-flatliner into data_classes

Commit:
update tests

Commit:
update tests

MichaelClifford commented:
@durandom @4n4nd Please review this PR and let me know if you have any other comments. Thanks!

Commit:
Add PromMetrics Observable class to stream historical data

Rename old PromMetrics class to PromMetricsLive (For live data stream).
Update app.py to use PromMetrics class when the required env variables are set.

durandom committed:
Merge pull request #39 from 4n4nd/master

Add PromMetrics Observable class to stream historical data

durandom reviewed (approved):
LGTM, but you need to rebase on top of master and run `pipenv install` again, to fix the conflicts in `Pipfile.lock`

Commit:
fix conflicts

4n4nd commented on metrics/__init__.py:
```diff
@@ -1,2 +1,3 @@
 from .filemetrics import FileMetrics
 from .prommetrics import PromMetrics
+from .prommetricslive import PromMetricsLive
```
Should I just comment this line?

durandom replied:
I think we dont need to import anything here. maybe we can completly empty this file
